src/Microsoft.Diagnostics.Monitoring.RestServer/DiagnosticServices.cs

```diff
@@ -22,27 +23,24 @@ internal sealed class DiagnosticServices : IDiagnosticServices
         // on a Windows operating system.
         private const string ProcessOperatingSystemWindowsValue = "windows";
 
-        // A Docker container's entrypoint process ID is 1
-        private static readonly ProcessKey DockerEntrypointProcessFilter = new ProcessKey(1);
-
-        // The amount of time to wait when checking if the docker entrypoint process is a .NET process
-        // with a diagnostics transport connection.
-        private static readonly TimeSpan DockerEntrypointWaitTimeout = TimeSpan.FromMilliseconds(250);
         // The amount of time to wait before cancelling get additional process information (e.g. getting
         // the process command line if the IEndpointInfo doesn't provide it).
         private static readonly TimeSpan ExtendedProcessInfoTimeout = TimeSpan.FromMilliseconds(500);
 
         private readonly IEndpointInfoSourceInternal _endpointInfoSource;
         private readonly CancellationTokenSource _tokenSource = new CancellationTokenSource();
         private readonly StorageOptions _storageOptions;
+        private readonly ProcessFilterOptions _defaultProcessOptions;
 
-        public DiagnosticServices(IEndpointInfoSource endpointInfoSource, IOptions<StorageOptions> storageOptions)
+        public DiagnosticServices(IEndpointInfoSource endpointInfoSource, IOptions<StorageOptions> storageOptions,
+            IOptions<ProcessFilterOptions> defaultProcessOptions)
```

Comment:
Without reloadability, using RuntimeId in the process filter is probably not going to work very well.

Reply:
I will add reload support here, but I really wouldn't expect anyone to use runtime id in configuration.

Reply:
If we don't think anyone is going to use runtime ID in configuration, then let's not offer it, especially since it has some (minor) special casing for it.

Reply:
What is the special casing for it?

Reply:
`if ((filter.Key == ProcessFilterKey.RuntimeId) && (processInfo.EndpointInfo.RuntimeInstanceCookie != Guid.Empty))`

If that wasn't there, then the whole method could be a switch rather that if-if-if-...